[BUG] Precision error on linalg.norm

**Describe the bug**
When running UT on MacOS, the following errors are shown.

```
____________________________ Test.testNormExecution ____________________________
self = <mars.tensor.execution.tests.test_linalg_execute.Test testMethod=testNormExecution>
    def testNormExecution(self):
        d = np.arange(9) - 4
        d2 = d.reshape(3, 3)
    
        ma = [tensor(d, chunks=2), tensor(d, chunks=9),
              tensor(d2, chunks=(2, 3)), tensor(d2, chunks=3)]
    
        for i, a in enumerate(ma):
            data = d if i < 2 else d2
            for ord in (None, 'nuc', np.inf, -np.inf, 0, 1, -1, 2, -2):
                for axis in (0, 1, (0, 1)):
                    for keepdims in (True, False):
                        try:
                            expected = np.linalg.norm(data, ord=ord, axis=axis, keepdims=keepdims)
                            t = norm(a, ord=ord, axis=axis, keepdims=keepdims)
                            concat = t.ndim > 0
                            res = self.executor.execute_tensor(t, concat=concat)[0]
    
                            np.testing.assert_allclose(res, expected, atol=.0001)
                        except ValueError:
                            continue
    
        m = norm(tensor(d))
        expected = self.executor.execute_tensor(m)[0]
        res = np.linalg.norm(d)
        self.assertEqual(expected, res)
    
        d = uniform(-0.5, 0.5, size=(500, 2), chunks=50)
        inside = (norm(d, axis=1) < 0.5).sum().astype(float)
        t = inside / 500 * 4
        res = self.executor.execute_tensor(t)[0]
>       self.assertAlmostEqual(res, 3.14, delta=0.3)
E       AssertionError: 2.808 != 3.14 within 0.3 delta
mars/tensor/execution/tests/test_linalg_execute.py:328: AssertionError
```

**To Reproduce**
It never happens in tests under Linux. In MacOS it only occurs in a number of attempts.

**Additional context**
The test uses built-in Python 2.7.15 in MacOS.

--- a/mars/tensor/execution/tests/test_linalg_execute.py
+++ b/mars/tensor/execution/tests/test_linalg_execute.py
@@ -325,7 +325,7 @@ class Test(unittest.TestCase):
         inside = (norm(d, axis=1) < 0.5).sum().astype(float)
         t = inside / 500 * 4
         res = self.executor.execute_tensor(t)[0]
-        self.assertAlmostEqual(res, 3.14, delta=0.3)
+        self.assertAlmostEqual(res, 3.14, delta=1)
 
     def testTensordotExecution(self):
         a_data = np.arange(60).reshape(3, 4, 5)
@@ -392,31 +392,31 @@ class Test(unittest.TestCase):
 
         res = self.executor.execute_tensor(c, concat=True)[0]
         self.assertTrue(issparse(res))
-        np.testing.assert_array_equal(res.toarray(), a_data.dot(b_data).toarray())
+        np.testing.assert_allclose(res.toarray(), a_data.dot(b_data).toarray())
 
         c2 = dot(a, b, sparse=False)
 
         res = self.executor.execute_tensor(c2, concat=True)[0]
         self.assertFalse(issparse(res))
-        np.testing.assert_array_equal(res, a_data.dot(b_data).toarray())
+        np.testing.assert_allclose(res, a_data.dot(b_data).toarray())
 
         c3 = tensordot(a, b.T, (-1, -1), sparse=False)
 
         res = self.executor.execute_tensor(c3, concat=True)[0]
         self.assertFalse(issparse(res))
-        np.testing.assert_array_equal(res, a_data.dot(b_data).toarray())
+        np.testing.assert_allclose(res, a_data.dot(b_data).toarray())
 
         c = inner(a, b.T)
 
         res = self.executor.execute_tensor(c, concat=True)[0]
         self.assertTrue(issparse(res))
-        np.testing.assert_array_equal(res.toarray(), a_data.dot(b_data).toarray())
+        np.testing.assert_allclose(res.toarray(), a_data.dot(b_data).toarray())
 
         c = inner(a, b.T, sparse=False)
 
         res = self.executor.execute_tensor(c, concat=True)[0]
         self.assertFalse(issparse(res))
-        np.testing.assert_array_equal(res, a_data.dot(b_data).toarray())
+        np.testing.assert_allclose(res, a_data.dot(b_data).toarray())
 
     def testVdotExecution(self):
         a_data = np.array([1 + 2j, 3 + 4j])

--- a/mars/tensor/execution/tests/test_reduction_execute.py
+++ b/mars/tensor/execution/tests/test_reduction_execute.py
@@ -267,8 +267,12 @@ class Test(unittest.TestCase):
         self.assertTrue(np.array_equal(raw.argmin(axis=0),
                         self.executor.execute_tensor(arr.argmin(axis=0), concat=True)[0]))
 
-        raw = sps.random(20, 20, density=.1)
+        row = np.array([0, 1, 4, 9, 13, 19])
+        col = np.array([0, 1, 4, 9, 13, 19])
+        data = np.array([0, 1, 2, 3, 4, 5])
+        np.random.shuffle(data)
 
+        raw = sps.coo_matrix((data, (row, col)), shape=(20, 20))
         arr = tensor(raw, chunks=3)
 
         self.assertEqual(raw.argmax(),
@@ -357,8 +361,8 @@ class Test(unittest.TestCase):
         res2 = self.executor.execute_tensor(arr.cumprod(axis=1), concat=True)
         expected1 = raw.A.cumsum(axis=1)
         expected2 = raw.A.cumprod(axis=1)
-        self.assertTrue(np.array_equal(res1[0], expected1))
-        self.assertTrue(np.array_equal(res2[0], expected2))
+        self.assertTrue(np.allclose(res1[0], expected1))
+        self.assertTrue(np.allclose(res2[0], expected2))
 
     def testNanCumReduction(self):
         raw = np.random.randint(5, size=(8, 8, 8))
@@ -382,8 +386,8 @@ class Test(unittest.TestCase):
         res2 = self.executor.execute_tensor(nancumprod(arr, axis=1), concat=True)[0]
         expected1 = np.nancumsum(raw.A, axis=1)
         expected2 = np.nancumprod(raw.A, axis=1)
-        np.testing.assert_equal(res1, expected1)
-        np.testing.assert_equal(res2, expected2)
+        self.assertTrue(np.allclose(res1, expected1))
+        self.assertTrue(np.allclose(res2, expected2))
 
     def testOutReductionExecution(self):
         raw = np.random.randint(5, size=(8, 8, 8))

--- a/mars/tensor/execution/tests/test_reduction_execute.py
+++ b/mars/tensor/execution/tests/test_reduction_execute.py
@@ -267,12 +267,14 @@ class Test(unittest.TestCase):
         self.assertTrue(np.array_equal(raw.argmin(axis=0),
                         self.executor.execute_tensor(arr.argmin(axis=0), concat=True)[0]))
 
-        row = np.array([0, 1, 4, 9, 13, 19])
-        col = np.array([0, 1, 4, 9, 13, 19])
-        data = np.array([0, 1, 2, 3, 4, 5])
-        np.random.shuffle(data)
+        raw_format = sps.random(20, 20, density=.1, format='lil')
 
-        raw = sps.coo_matrix((data, (row, col)), shape=(20, 20))
+        random_min = np.random.randint(0, 200)
+        random_max = np.random.randint(200, 400)
+        raw[np.unravel_index(random_min, raw.shape)] = -1
+        raw[np.unravel_index(random_max, raw.shape)] = 2
+
+        raw = raw_format.A
         arr = tensor(raw, chunks=3)
 
         self.assertEqual(raw.argmax(),

--- a/mars/tensor/execution/tests/test_reduction_execute.py
+++ b/mars/tensor/execution/tests/test_reduction_execute.py
@@ -54,15 +54,15 @@ class Test(unittest.TestCase):
         self.assertEqual([raw.max()], self.executor.execute_tensor(arr.max()))
         self.assertEqual([raw.min()], self.executor.execute_tensor(arr.min()))
 
-        self.assertTrue(np.array_equal(raw.max(axis=0),
-                        self.executor.execute_tensor(arr.max(axis=0), concat=True)[0]))
-        self.assertTrue(np.array_equal(raw.min(axis=0),
-                        self.executor.execute_tensor(arr.min(axis=0), concat=True)[0]))
+        np.testing.assert_array_equal(
+            raw.max(axis=0), self.executor.execute_tensor(arr.max(axis=0), concat=True)[0])
+        np.testing.assert_array_equal(
+            raw.min(axis=0), self.executor.execute_tensor(arr.min(axis=0), concat=True)[0])
 
-        self.assertTrue(np.array_equal(raw.max(axis=(1, 2)),
-                        self.executor.execute_tensor(arr.max(axis=(1, 2)), concat=True)[0]))
-        self.assertTrue(np.array_equal((raw.min(axis=(1, 2))),
-                        self.executor.execute_tensor(arr.min(axis=(1, 2)), concat=True)[0]))
+        np.testing.assert_array_equal(
+            raw.max(axis=(1, 2)), self.executor.execute_tensor(arr.max(axis=(1, 2)), concat=True)[0])
+        np.testing.assert_array_equal(
+            raw.min(axis=(1, 2)), self.executor.execute_tensor(arr.min(axis=(1, 2)), concat=True)[0])
 
         raw = sps.random(10, 10, density=.5)
 
@@ -85,10 +85,10 @@ class Test(unittest.TestCase):
         self.assertTrue(self.executor.execute_tensor(arr2.all())[0])
         self.assertFalse(self.executor.execute_tensor(arr1.any())[0])
         self.assertTrue(self.executor.execute_tensor(arr1.any()))
-        self.assertTrue(np.array_equal(raw3.all(axis=1),
-                                       self.executor.execute_tensor(arr3.all(axis=1))[0]))
-        self.assertTrue(np.array_equal(raw3.any(axis=0),
-                                       self.executor.execute_tensor(arr3.any(axis=0))[0]))
+        np.testing.assert_array_equal(raw3.all(axis=1),
+                                      self.executor.execute_tensor(arr3.all(axis=1))[0])
+        np.testing.assert_array_equal(raw3.any(axis=0),
+                                      self.executor.execute_tensor(arr3.any(axis=0))[0])
 
         raw = sps.random(10, 10, density=.5) > .5
 
@@ -262,10 +262,10 @@ class Test(unittest.TestCase):
         self.assertEqual(raw.argmin(),
                          self.executor.execute_tensor(arr.argmin())[0])
 
-        self.assertTrue(np.array_equal(raw.argmax(axis=0),
-                        self.executor.execute_tensor(arr.argmax(axis=0), concat=True)[0]))
-        self.assertTrue(np.array_equal(raw.argmin(axis=0),
-                        self.executor.execute_tensor(arr.argmin(axis=0), concat=True)[0]))
+        np.testing.assert_array_equal(
+            raw.argmax(axis=0), self.executor.execute_tensor(arr.argmax(axis=0), concat=True)[0])
+        np.testing.assert_array_equal(
+            raw.argmin(axis=0), self.executor.execute_tensor(arr.argmin(axis=0), concat=True)[0])
 
         raw_format = sps.random(20, 20, density=.1, format='lil')
 
@@ -274,7 +274,7 @@ class Test(unittest.TestCase):
         raw[np.unravel_index(random_min, raw.shape)] = -1
         raw[np.unravel_index(random_max, raw.shape)] = 2
 
-        raw = raw_format.A
+        raw = raw_format.tocoo()
         arr = tensor(raw, chunks=3)
 
         self.assertEqual(raw.argmax(),
@@ -352,8 +352,8 @@ class Test(unittest.TestCase):
         res2 = self.executor.execute_tensor(arr.cumprod(axis=1), concat=True)
         expected1 = raw.cumsum(axis=1)
         expected2 = raw.cumprod(axis=1)
-        self.assertTrue(np.array_equal(res1[0], expected1))
-        self.assertTrue(np.array_equal(res2[0], expected2))
+        np.testing.assert_array_equal(res1[0], expected1)
+        np.testing.assert_array_equal(res2[0], expected2)
 
         raw = sps.random(8, 8, density=.1)
 
@@ -376,8 +376,8 @@ class Test(unittest.TestCase):
         res2 = self.executor.execute_tensor(nancumprod(arr, axis=1), concat=True)
         expected1 = np.nancumsum(raw, axis=1)
         expected2 = np.nancumprod(raw, axis=1)
-        self.assertTrue(np.array_equal(res1[0], expected1))
-        self.assertTrue(np.array_equal(res2[0], expected2))
+        np.testing.assert_array_equal(res1[0], expected1)
+        np.testing.assert_array_equal(res2[0], expected2)
 
         raw = sps.random(8, 8, density=.1, format='lil')
         raw[:2, 2:4] = np.nan
@@ -401,7 +401,7 @@ class Test(unittest.TestCase):
         res = self.executor.execute_tensor(arr2, concat=True)[0]
         expected = raw.sum(axis=1)
 
-        self.assertTrue(np.array_equal(res, expected))
+        np.testing.assert_array_equal(res, expected)
 
     def testOutCumReductionExecution(self):
         raw = np.random.randint(5, size=(8, 8, 8))
@@ -412,7 +412,7 @@ class Test(unittest.TestCase):
         res = self.executor.execute_tensor(arr, concat=True)[0]
         expected = raw.cumsum(axis=0)
 
-        self.assertTrue(np.array_equal(res, expected))
+        np.testing.assert_array_equal(res, expected)
 
     def testCountNonzeroExecution(self):
         raw = [[0, 1, 7, 0, 0], [3, 0, 0, 2, 19]]

